keanu-project/src/test/java/io/improbable/keanu/vertices/bool/nonprobabilistic/operators/multiple/BoolConcatenationVertexTest.java

```diff
@@ -28,38 +29,38 @@ public void canConcatVectorsOfDifferentSize() {
         ConstantBoolVertex a = new ConstantBoolVertex(new boolean[]{true, true, true});
         ConstantBoolVertex b = new ConstantBoolVertex(new boolean[]{false, false, false, false, false, false});
 
-        BoolConcatenationVertex concatZero = new BoolConcatenationVertex(1, a, b);
+        BoolConcatenationVertex concatZero = new BoolConcatenationVertex(0, a, b);
 
-        Assert.assertArrayEquals(new long[]{1, 9}, concatZero.getShape());
+        Assert.assertArrayEquals(new long[]{9}, concatZero.getShape());
         Assert.assertArrayEquals(new double[]{1, 1, 1, 0, 0, 0, 0, 0, 0}, concatZero.getValue().asFlatDoubleArray(), 0.001);
     }
 
     @Test
     public void canConcatScalarToVector() {
         ConstantBoolVertex a = new ConstantBoolVertex(new boolean[]{true, true, true});
-        ConstantBoolVertex b = new ConstantBoolVertex(false);
+        ConstantBoolVertex b = new ConstantBoolVertex(new boolean[]{false});
```

Comment:
Ah, so this is now a subtle difference - creating a scalar vs a vector. After thinking about it for a while, I think I like the semantics.